client/src/components/BlogPost.js

```diff
@@ -18,14 +20,20 @@ const styles = {
 function BlogPost({ allPosts, match }) {
   const post = allPosts.find(post => post.id === match.params.id)
   return (
-    <div className="app__div box box--large" style={styles}>
-      <h1>{post.title}</h1>
-      <figure>
-        <img className="app__img" src={post.featuredImg.src} />
-        <figcaption>{post.featuredImg.caption}</figcaption>
-      </figure>
-      <div>{post.body}</div>
-    </div>
+    <ErrorBoundary 
+      statusCode={404} 
+      message={`Sorry, we couldn't find "${post.tile}" in our records.`}
+    >
+      <div className="app__div box box--large" style={styles}>
+        <h1>{post.title}</h1>
+        <figure>
+          <img className="app__img" src={post.featuredImg.src} />
+          <figcaption>{post.featuredImg.caption}</figcaption>
+        </figure>
+        <div>{post.body}</div>
+      </div>
+    </ErrorBoundary>
```

Comment:
https://github.com/Brandon-Pampuch/Easypeasy-blog/blob/a871bcddcde5cfc0446e7a2f283750ba7071eb0c/client/src/components/BlogPost.js#L23-L35

I wish the app I received had this level of thoughtful and professional error handling.  💯